presto-native-execution/presto_cpp/main/SystemConnector.cpp

```diff
@@ -0,0 +1,297 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/SystemConnector.h"
+#include "presto_cpp/main/PrestoTask.h"
+#include "presto_cpp/main/TaskManager.h"
+
+namespace facebook::presto {
+
+using namespace velox;
+
+SystemTableHandle::SystemTableHandle(
+    std::string connectorId,
+    std::string schemaName,
+    std::string tableName)
+    : ConnectorTableHandle(std::move(connectorId)),
+      schemaName_(std::move(schemaName)),
+      tableName_(std::move(tableName)) {
+  VELOX_USER_CHECK_EQ(
+      schemaName_, "runtime", "SystemConnector supports only runtime schema");
+  VELOX_USER_CHECK_EQ(
+      tableName_, "tasks", "SystemConnector supports only tasks table");
+
+  std::vector<std::string> kTaskColumnNames = {
+      "node_id",
+      "task_id",
+      "stage_execution_id",
+      "stage_id",
+      "query_id",
+      "state",
+      "splits",
+      "queued_splits",
+      "running_splits",
+      "completed_splits",
+      "split_scheduled_time_ms",
+      "split_cpu_time_ms",
+      "split_blocked_time_ms",
+      "raw_input_bytes",
+      "raw_input_rows",
+      "processed_input_bytes",
+      "processed_input_rows",
+      "output_bytes",
+      "output_rows",
+      "physical_written_bytes",
+      "created",
+      "start",
+      "last_heartbeat",
+      "end"};
+  std::vector<velox::TypePtr> kTaskColumnTypes = {
+      velox::VARCHAR(),   velox::VARCHAR(),   velox::VARCHAR(),
+      velox::VARCHAR(),   velox::VARCHAR(),   velox::VARCHAR(),
+      velox::BIGINT(),    velox::BIGINT(),    velox::BIGINT(),
+      velox::BIGINT(),    velox::BIGINT(),    velox::BIGINT(),
+      velox::BIGINT(),    velox::BIGINT(),    velox::BIGINT(),
+      velox::BIGINT(),    velox::BIGINT(),    velox::BIGINT(),
+      velox::BIGINT(),    velox::BIGINT(),    velox::TIMESTAMP(),
+      velox::TIMESTAMP(), velox::TIMESTAMP(), velox::TIMESTAMP()};
+  kTaskSchema_ = ROW(std::move(kTaskColumnNames), std::move(kTaskColumnTypes));
+}
+
+std::string SystemTableHandle::toString() const {
+  return fmt::format("schema: {} table: {}", schemaName_, tableName_);
+}
+
+SystemDataSource::SystemDataSource(
+    const std::shared_ptr<const RowType>& outputType,
+    const std::shared_ptr<connector::ConnectorTableHandle>& tableHandle,
+    const std::unordered_map<
+        std::string,
+        std::shared_ptr<connector::ColumnHandle>>& columnHandles,
+    const TaskManager* taskManager,
+    velox::memory::MemoryPool* FOLLY_NONNULL pool)
+    : taskManager_(taskManager), pool_(pool) {
+  auto systemTableHandle =
+      std::dynamic_pointer_cast<SystemTableHandle>(tableHandle);
+  VELOX_CHECK_NOT_NULL(
+      systemTableHandle,
+      "TableHandle must be an instance of SystemTableHandle");
+  VELOX_USER_CHECK_EQ(
```

Comment:
These checks are redundant as they duplicate checks in SystemTableHandle's constructor.

Reply:
Yeah removed these as well.